[llm] upgrade vllm to 0.12.0

kouroshHakha opened the pull request:
Related prs that we should review when upgrading fully: 
- https://github.com/ray-project/ray/pull/58820
- Note from Rui: When we bump new vllm version, we should go with 0.11.2 instead of 0.11.1, which fixes a Ray multi-node PP regression that was introduced when adding torch-based PP
https://github.com/vllm-project/vllm/releases/tag/v0.11.2

Issues:
- closes https://github.com/ray-project/ray/issues/58937
- closes https://github.com/ray-project/ray/issues/58973
- closes https://github.com/ray-project/ray/issues/58702

kouroshHakha committed:
wip

Signed-off-by: Kourosh Hakhamaneshi <Kourosh@anyscale.com>

eicherseiji committed:
Bump vLLM, pydantic, packaging deps

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Merge branch 'master' of https://github.com/ray-project/ray into kh/temp-vllm-0.12.0

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Bump to 0.11.2

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

nrghosh committed:
Merge remote-tracking branch 'upstream/master' into kh/temp-vllm-0.12.0

# Conflicts:
#	python/deplocks/llm/rayllm_py311_cpu.lock
#	python/deplocks/llm/rayllm_py311_cu128.lock
#	python/deplocks/llm/rayllm_test_py311_cpu.lock
#	python/deplocks/llm/rayllm_test_py311_cu128.lock

nrghosh committed:
Merge branch 'master' into kh/temp-vllm-0.12.0

nrghosh committed:
recompile deps + fix lock files

Signed-off-by: Nikhil Ghosh <nikhil@anyscale.com>

nrghosh committed:
Fix TypeError: argparse.Namespace() got multiple values for keyword argument 'tokens_only'

Addresses https://github.com/ray-project/ray/issues/58973

- vLLM release 0.11.1 introduces tokens_only arguments to both
  FrontendArgs and EngineArgs. VLLMEngine.start() gathers arguments from
both of them, which raises errors when collisions occur
- Allow different argument sets to define the same arguments by name,
  and give precedence to the engine args (in case of collisions), then
merge the dicts

Signed-off-by: Nikhil Ghosh <nikhil@anyscale.com>

nrghosh committed:
Merge branch 'master' into kh/temp-vllm-0.12.0

nrghosh committed:
build / refresh all lock files

- fixing ci error with `//ci/raydepsets:raydepsets -- build
--all-configs`

Signed-off-by: Nikhil Ghosh <nikhil@anyscale.com>

nrghosh reviewed (commented):
release tests all passing except for `[jailed]llm_batch_vllm_multi_node (None) (0)` - tracked here https://github.com/ray-project/ray/issues/58062#issuecomment-3439259664

<img width="356" height="340" alt="image" src="https://github.com/user-attachments/assets/2bda241e-c70c-4c76-a10d-63b621434849" />


PR to fix (by Rui) - https://github.com/ray-project/ray/pull/58866

nrghosh reviewed (commented):
`python/ray/llm/tests/batch/gpu
/processor/test_vllm_engine_proc.py::test_embedding_model` is failing due to 

```RuntimeError: flashinfer-cubin version (0.5.3) does not match flashinfer version (0.5.2)```

there is a mismatch between flashinfer-python (0.5.2) and flashinfer-cubin (0.5.3). The lock file pins flashinfer-python==0.5.2, but then vLLM 0.11.2 pulls in flashinfer-cubin==0.5.3 as a transitive dependency.

nrghosh committed:
constrain flashinfer-cubin to 0.5.2 to match flashinfer-python version required by vllm 0.11.2

Signed-off-by: Nikhil Ghosh <nikhil@anyscale.com>

nrghosh committed:
update lock files for flashinfer-cubin pin

Signed-off-by: Nikhil Ghosh <nikhil@anyscale.com>

nrghosh reviewed (commented):
test failure for `/doc:source/data/doc_code/working-with-llms/embedding_example ` 

```
msgspec.ValidationError: Expected `bool`, got `None` - at `$[4][10]`
```

issue is in vllm - fixed here (merged): https://github.com/vllm-project/vllm/pull/29364 - so pushing to 0.12.0

nrghosh committed:
Merge branch 'master' into kh/temp-vllm-0.12.0

Signed-off-by: Nikhil G <nrghosh@users.noreply.github.com>

nrghosh committed:
bump to 0.12.0

Signed-off-by: Nikhil Ghosh <nikhil@anyscale.com>

elliot-barn committed:
updating lock files

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn committed:
working depset

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

nrghosh committed:
Merge branch 'master' into kh/temp-vllm-0.12.0

Signed-off-by: Nikhil G <nrghosh@users.noreply.github.com>

nrghosh committed:
fix imports - vllm modules moved around

Signed-off-by: Nikhil Ghosh <nikhil@anyscale.com>

nrghosh committed:
Last API change fixes

vLLM renamed guided_decoding to structured_outputs and changed
the embedding API:

- SamplingParams: GuidedDecodingParams -> StructuredOutputsParams,
  guided_decoding -> structured_outputs (vllm-project/vllm#22772,
  vllm-project/vllm#29326)

- Embedding: use encode(pooling_params=...) instead of
  generate(sampling_params=...) for pooling tasks
  (vllm-project/vllm#16188, vllm-project/vllm#25524)

- EngineArgs: guided_decoding_backend -> structured_outputs_config

User-facing "guided_decoding" key in sampling_params dict preserved
for backwards compatibility.

Signed-off-by: Nikhil Ghosh <nikhil@anyscale.com>

nrghosh committed:
Merge branch 'master' into kh/temp-vllm-0.12.0

nrghosh reviewed (commented):
after this merges, will undo transformers pin 

ctx: https://github.com/ray-project/ray/pull/58980#pullrequestreview-3506711143

> [eicherseiji](https://github.com/eicherseiji) left a comment
Ideally we can drop the transformers pin in the next vLLM version bump

cc @eicherseiji the multi-gpu tests (dp_pd_example and dp_basic_example.py) are OOMing on warmup (at first glance) - is reducing sequence length ok, or did something change with the compute configs? They both run on this branch with on a workspace w/ 4xL4.

nrghosh committed:
Reduce memory usage in multi-GPU DP doc tests

Signed-off-by: Nikhil Ghosh <nikhil@anyscale.com>

nrghosh reviewed (approved):
looks good @kouroshHakha 

<img width="366" height="95" alt="Image" src="https://github.com/user-attachments/assets/f3734cee-eae4-46bd-bd2b-2ffc171e80ae" />

richardliaw reviewed (approved):
stamp
